sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -68,6 +68,18 @@ class TypedColumn[-T, U](
     }
     new TypedColumn[T, U](newExpr, encoder)
   }
+
+  /**
+   * Gives the TypedColumn a name (alias).
+   * If the current TypedColumn has metadata associated with it, this metadata will be propagated
+   * to the new column.
+   *
+   * @group expr_ops
+   * @since 2.0.0
+   */
+  override def name(alias: String): TypedColumn[T, U] =
```

Comment:
looks like `def as(alias: Symbol): Column` don't delegate to `name`, we should change it.

Can you go through all alias methods of `Column` and make sure they all delegate to `name` or re-implemented in `TypedColumn`?